fastlane/lanes/build.rb

```diff
@@ -127,18 +127,24 @@
   # Builds the WordPress app and uploads it to TestFlight, for beta-testing or final release
   #
   # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
-  # @option [Boolean] skip_prechecks (default: false) If true, don't run the ios_build_prechecks and ios_build_preflight
+  # @option [Boolean] skip_prechecks (default: false) If true, don't run the prechecks and ios_build_preflight
   # @option [Boolean] create_release If true, creates a GitHub Release draft after the upload, with zipped xcarchive as artefact
   # @option [Boolean] beta_release If true, the GitHub release will be marked as being a pre-release
   #
   # @called_by CI
   #
   desc 'Builds and uploads for distribution to App Store Connect'
   lane :build_and_upload_app_store_connect do |options|
-    ios_build_prechecks(skip_confirm: options[:skip_confirm], external: true) unless options[:skip_prechecks]
-    ios_build_preflight unless options[:skip_prechecks]
+    unless options[:skip_prechecks]
+      ensure_git_status_clean unless is_ci
+      ios_build_preflight
+    end
+
+    UI.important("Building version #{release_version_current} (#{build_code_current}) and uploading to TestFlight")
+    UI.user_error!('Aborted by user request') if !options[:skip_confirm] && !UI.confirm('Do you want to continue?')
```

Comment:
What do you think of using positive logic in the condition?

```suggestion
    UI.user_error!('Aborted by user request') unless options[:skip_confirm] || UI.confirm('Do you want to continue?')
```

This assumes that the second part of the `||` (`UI.confirm`) will run only if the first (`:skip_confirm`) is `false`. I think that's what most languages do for optimization purposes, but I haven't checked here.

Reply:
Agreed, I like that change. I updated all the occurrences in https://github.com/wordpress-mobile/WordPress-iOS/commit/4ae879d7698d87f88daec36f04a022ad2770c871

Reply:
Looks like this entry was missed in 4ae879d:

![image](https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/458ee207-99f6-49a1-8961-f481ed15259d)

Reply:
```suggestion
    UI.user_error!('Aborted by user request') unless options[:skip_confirm] || UI.confirm('Do you want to continue?')
```

Reply:
(I had to unresolved the conversation to apply the suggestion)

Reply:
Ah thank you for catching those!